py/foundation/entity.py

```diff
@@ -71,31 +38,25 @@ def words(self) -> Iterable['Word']:
 
     STRONGLY RECOMMENDED not to override this.
     """
-    yield from chain.from_iterable(e.words() for e in self.children)
+    yield from chain.from_iterable(E.words() for E in self.children)
+
+  @property
+  def entity_text(self) -> Optional[str]:
+    entity_fields = (field.name for field in fields(self))
```

Comment:
Why do you we need to do this instead of just self.text? Sorry I am unfamiliar with some of these specific things about dataclasses

Reply:
So this is something I've been wrestling with today/yesterday...if you have a better way I'd love to hear it because I'm still not happy with this. The short answer is that not all Entity subclasses have a `text` field.

I want there to be a way to get "text" from an Entity, regardless of if that class has a `text` field, hence this `entity_text` property. Another option would be to have `text: Optional[str]` on all Entities, but I don't like that because it shouldn't be optional for Word, Phrase, etc.

The reason I don't just have a text property on all that gives `None` for some and always strings for Words etc., is that for the serializing code, we want `text` to be one of the dataclass fields for the Entity types that need it, (so that we have Word come out as `{text: 'whatever', type: 'Word'...}`).

Anyways, I'm likely overcomplicating this, but I have yet to come up with a clean way and would love input.